core/modules/czar/CzarConfig.h

```diff
@@ -0,0 +1,133 @@
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_QSERV_CZAR_CZARCONFIG_H
+#define LSST_QSERV_CZAR_CZARCONFIG_H
+
+// System headers
+
+// Qserv headers
+#include "css/CssConfig.h"
+#include "mysql/MySqlConfig.h"
+#include "util/ConfigStore.h"
+
+namespace lsst {
+namespace qserv {
+namespace czar {
+
+/**
+ *  Provide all configuration parameters for a Qserv Czar instance
+ *
+ *  Parse an INI configuration file, identify required parameters and ignore
+ *  others, analyze and store them inside private member variables, use default
+ *  values for missing parameters, provide accessor for each of these variable.
+ *  This class hide configuration complexity from other part of the code.
+ *  All private member variables are related to Czar parameters and are immutables.
+ *
+ */
+class CzarConfig {
+public:
+    CzarConfig(std::string configFileName)
+        : CzarConfig(util::ConfigStore(configFileName)) {
+    }
+
+    CzarConfig(CzarConfig const&) = delete;
+    CzarConfig& operator=(CzarConfig const&) = delete;
+
+    /** Overload output operator for current class
+     *
+     * @param out
+     * @param workerConfig
+     * @return an output stream
+     */
+    friend std::ostream& operator<<(std::ostream &out, CzarConfig const& czarConfig);
+
+    /* Get MySQL configuration for czar MySQL result database
+     *
+     * @return a structure containing MySQL parameters
+     */
+    mysql::MySqlConfig const& getMySqlResultConfig() const {
+        return _mySqlResultConfig;
+    }
+
+    std::string const& getLogConfig() const {
+        return _logConfig;
+    }
+
+    /* Get MySQL configuration for czar MySQL qmeta database
+     *
+     * @return a structure containing MySQL parameters
+     */
+    mysql::MySqlConfig const& getMySqlQmetaConfig() const {
+        return _mySqlQmetaConfig;
+    }
+
+    /* Get CSS parameters as a collection of key-value
+     *
+     * Do not check CSS parameters consistency
+     *
+     * @return a structure containing CSS parameters
+     */
+    std::map<std::string, std::string> const& getCssConfigMap() const {
+        return _cssConfigMap;
+    }
+
+    /* Get path to directory where the empty chunk files resides
+     *
+     * Each empty chunk file is related to one cosmic dataset
+     *
+     * @return path to directory where the empty chunk files resides
+     */
+    std::string const& getEmptyChunkPath() const {
+        return _emptyChunkPath;
+    }
+
+    /* Get hostname and port for xrootd manager
+     *
+     * "localhost:1094" is the most reasonable default, even though it is
+     * the wrong choice for all but small developer installations
+     *
+     * @return a string containing "<hostname>:<port>"
+     */
+    std::string const& getXrootdFrontendUrl() const {
+        return _xrootdFrontendUrl;
+    }
+
+private:
+
+    CzarConfig(util::ConfigStore const& ConfigStore);
```

Comment:
If it is not used by anything else but public constructor then you can probably do everything inside public constructor body. Why do we need two constructors when one is enough?